.github/workflows/publish-pypi.yml

```diff
@@ -25,7 +25,7 @@ jobs:
           python-version: "3.11"
       - name: Setup for poetry
         run: |
-          python -m pip install poetry
+          python -m pip install poetry poetry-dynamic-versioning
```

Comment:
Is dynamic versioning not something automatic that poetry does? It is interesting to me that you have to specify it when installing poetry.

Reply:
Thanks for this comment - versioning is in general a task where developers make decisions about how to distinguish releases. There are a number of non-technical aspects like user needs assessment, contributor perspectives, and timing of release alongside other dependencies. This might make it challenging to distinguish automatically what and when to version things. Poetry by default provides this decision to developers via a [`poetry version`](https://python-poetry.org/docs/cli/#version) command which has to be issued manually.

Dynamic versioning for poetry is a relatively new plugin for poetry and implements some non-standard behavior. It also involves aspects of source control like git tags, which are opinionated and not necessarily in the span of control of poetry or python (which I believe might be some of the reasoning as to why it's not a default in poetry).